### PR TITLE
Run2-hcx144 Correct the active length calculation and its usage

### PR DESCRIPTION
**Pull request by bsunanda**
Active length in scintillator is used for MIP calculation.

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc [hunk 1]
```diff
@@ -67,7 +67,6 @@ class HcalHBHEMuonAnalyzer :  public edm::one::EDAnalyzer<edm::one::WatchRuns,ed
 
 public:
   explicit HcalHBHEMuonAnalyzer(const edm::ParameterSet&);
-  ~HcalHBHEMuonAnalyzer();
 
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
```
Comment (ianna) [inline Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc]: @bsunanda - please, apply the patch from the code checks:
```
diff --git a/Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc b/Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc
index afef29c..a471582 100644
--- a/Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc
+++ b/Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc
@@ -71,10 +71,10 @@ public:
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
 
 private:
-  virtual void beginJob() override;
-  virtual void analyze(edm::Event const&, edm::EventSetup const&) override;
-  virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
-  virtual void endRun(edm::Run const&, edm::EventSetup const&) override {}
+  void beginJob() override;
+  void analyze(edm::Event const&, edm::EventSetup const&) override;
+  void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  void endRun(edm::Run const&, edm::EventSetup const&) override {}
   virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {}
   virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {}
   void   clearVectors();
```

**Review by ianna — CHANGES_REQUESTED** [review]

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [inline Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc]: Done

**Comment by bsunanda** [comment]
@ianna @lpernie Could you approve this PR?

**Comment by ianna** [comment]
+1

**Comment by lpernie** [comment]
+1

File: Geometry/HcalCommonData/src/HcalDDDSimConstants.cc [hunk 2]
```diff
@@ -339,6 +339,22 @@ int HcalDDDSimConstants::getFrontLayer(const int det, const int eta) const {
   return lay;
 }
 
+int HcalDDDSimConstants::getLastLayer(const int det, const int eta) const {
+
+  int lay=0;
+  if (det == 1) {
+    if      (eta == 15 || eta == -15) lay = layBHB[1];
```
Comment (davidlange6) [inline Geometry/HcalCommonData/src/HcalDDDSimConstants.cc]: hi @bsunanda - why not if ( std::abs(eta)==15) (etc)

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @lpernie Could you approve this yet again - made changes as suggested by David

**Comment by ianna** [comment]
+1

File: Geometry/HcalCommonData/interface/HcalDDDRecConstants.h [hunk 3]
```diff
@@ -78,9 +79,9 @@ class HcalDDDRecConstants {
   HcalID                    getHCID(int subdet, int ieta, int iphi, int lay,
 				    int idepth) const;
   std::vector<HFCellParameters>    getHFCellParameters() const;
-  void                      getLayerDepth(int ieta, std::map<int,int>& layers) const;
-  int                       getLayerFront(int det, int eta, int phi, int depth) const;
-  double                    getLayer0Wt(int det, int phi, int zside) const {return hcons.getLayer0Wt(det,phi,zside);}
+  void                      getLayerDepth(const int ieta, std::map<int,int>& layers) const;
```
Comment (davidlange6) [inline Geometry/HcalCommonData/interface/HcalDDDRecConstants.h]: hi @bsunanda - why add the consts? They have no effect (as these are not references)

Comment (bsunanda) [inline Geometry/HcalCommonData/interface/HcalDDDRecConstants.h]: Could you merge the PR. I shall update this later on. The merged code is needed by many these days and is needed for the Barrel TDR studies

Comment (bsunanda) [inline Geometry/HcalCommonData/interface/HcalDDDRecConstants.h]: In fact there are other functions invoking this which require these consts - cannot be eliminated

Comment (bsunanda) [reply Geometry/HcalCommonData/interface/HcalDDDRecConstants.h]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ianna** [comment]
+1

**Comment by bsunanda** [comment]
@lpernie @franzoni Can you approve this PR

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @lpernie @arunhep @franzoni Please approve this PR

**Comment by lpernie** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**